packages/@aws-cdk/core/lib/bundling.ts

```diff
@@ -5,6 +5,22 @@ import { FileSystem } from './fs';
 import { dockerExec } from './private/asset-staging';
 import { quiet, reset } from './private/jsii-deprecated';
 
+/**
+ * Methods to build Docker CLI arguments for builds using secrets.
+ *
+ * Docker BuildKit must be enabled to use build secrets.
```

Comment:
Can we get a unit test that verifies that an error is thrown if BuildKit is not enabled?

Reply:
I looked into this but it seems it's not entirely trivial.

I considered pulling the value from environment variable `DOCKER_BUILDKIT` but this isn't reliable:

- Systems with Docker Desktop have BuildKit enabled by default and the environment variable is unset
- The setting could be enabled in `/etc/docker/daemon` which won't be reflected in the environment variable or be usable on a non-Linux host.

Further, I tried using `docker plugin ls` but this may behave differently depending on permissions, for example;

```
% docker plugin ls
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/plugins": dial unix /var/run/docker.sock: connect: permission denied
```

I have added a `@see` to the relevant code and readme sections.  I'm hopeful this is a valid compromise?

Reply:
Ok :+1: this is fine